File: solution_code/brute_force_tle/l.py
import sys
from collections import deque


def bfs_reach(graph: list[list[int]], starts: tuple[int, int, int]) -> int:
    seen = [False] * len(graph)
    dq = deque()
    for s in starts:
        if not seen[s]:
            seen[s] = True
            dq.append(s)
    count = 0
    while dq:
        u = dq.popleft()
        count += 1
        for v in graph[u]:
            if not seen[v]:
                seen[v] = True
                dq.append(v)
    return count


def solve() -> None:
    data = sys.stdin.buffer.read().split()
    if not data:
        return
    it = iter(data)
    t = int(next(it))
    out = []
    for _ in range(t):
        n = int(next(it))
        m = int(next(it))
        budget = int(next(it))
        fee = [int(next(it)) for _ in range(n)]
        graph = [[] for _ in range(n)]
        for _ in range(m):
            u = int(next(it)) - 1
            v = int(next(it)) - 1
            graph[u].append(v)
            graph[v].append(u)

        best = -1
        for i in range(n):
            for j in range(i + 1, n):
                for k in range(j + 1, n):
                    if fee[i] + fee[j] + fee[k] != budget:
                        continue
                    best = max(best, bfs_reach(graph, (i, j, k)))
        out.append(str(best))
    sys.stdout.write("\n".join(out))


if __name__ == "__main__":
    solve()

File: solution_code/brute_force_tle/l.cpp
#include <bits/stdc++.h>
using namespace std;

int bfs_reach(const vector<vector<int>>& g, const vector<int>& starts) {
    vector<int> vis((int)g.size(), 0);
    queue<int> q;
    for (int s : starts) {
        if (!vis[s]) {
            vis[s] = 1;
            q.push(s);
        }
    }
    int count = 0;
    while (!q.empty()) {
        int u = q.front();
        q.pop();
        ++count;
        for (int v : g[u]) {
            if (!vis[v]) {
                vis[v] = 1;
                q.push(v);
            }
        }
    }
    return count;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;
    while (t--) {
        int n, m;
        long long B;
        cin >> n >> m >> B;
        vector<long long> fee(n);
        for (int i = 0; i < n; ++i) cin >> fee[i];
        vector<vector<int>> g(n);
        for (int i = 0; i < m; ++i) {
            int u, v;
            cin >> u >> v;
            --u;
            --v;
            g[u].push_back(v);
            g[v].push_back(u);
        }

        int best = -1;
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                for (int k = j + 1; k < n; ++k) {
                    if (fee[i] + fee[j] + fee[k] != B) continue;
                    best = max(best, bfs_reach(g, {i, j, k}));
                }
            }
        }
        cout << best << '\n';
    }
}
